src/models/Settings.js

```diff
@@ -238,6 +247,57 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
           return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
+      },
+      configuredIdpsOptionsSet: {
+        deps: ['configuredSocialIdps'],
+        fn: function (configuredSocialIdps) {
+          _.each(configuredSocialIdps, function(idp) {
+            var type = idp.type.toLowerCase();
+            //set required properties
+            idp['attr'] = 'social-auth-' + type + '-button';
+            idp['className'] = 'social-auth-button ' + idp['attr'];
+            idp['title'] = Okta.loc('socialauth.' + type + '.label');
+            idp['click'] = socialAuthClickHandler;
+          })
+          return configuredSocialIdps;
+        },
+        cache: true
+      },
+
+      //custom buttons
+      customButtonsOptionsSet: {
+        deps: ['customButtons'],
+        fn: function (customButtons) {
+          _.each(customButtons, function(button) {
+            //add custom auth class name
+            button['className'] = button['className'] + ' default-custom-button';
```

Comment:
Do we need both the idp & customButtons derived properties? Can we just combine this in one? Also, agree, we need to whitelist the properties that are sent to us. Similar to the other comment - but create new objects here, don't modify the existing one they pass in. Doing this, we can just pick and choose (like @mauriciocastillosilva-okta suggested) what we want to take from it).